src/modelFactory/PluginAction.h

```diff
@@ -29,47 +28,57 @@
 #include "clang/Tooling/CommonOptionsParser.h"
 #include "clang/Tooling/Tooling.h"
 #include "ClangDiagnosticPrinter.h"
-
+#include "IModelFactory.h"
```

Comment:
This might be a reason speaking for the template solution. This is 'loosely' coupled to ModelFactory via the interface. The template solution was completely independent.